### PR TITLE
Remove Keepalived/DRBD based HA

### PR DESCRIPTION
**Pull request by markan**
### Description

The DRBD/keepalived based HA subsystem was deprecated as of Chef Server
12.9, and officially reached end of life on 2019-03-31. It will be
disabled in Chef Server 13.

This blocks off usage of the ha commands in chef-server-ctl, and
prevents reconfigure from supporting the ha topology.

There's a lot more work to do to complete cleanup, but this removes
key functionality in preparation for further work.

Signed-off-by: Mark Anderson <mark@chef.io>

### Check List

- [x] New functionality includes tests
- [x] All tests pass
- [x] All commits have been signed-off for the Developer Certificate of Origin. See <https://github.com/chef/chef/blob/master/CONTRIBUTING.md#developer-certification-of-origin-dco>

### CODE REVIEW

**Comment by markan** [comment]
@chef/docs-team:

You can ignore most of this PR (just code cleanups), but there are two error messages I'd love a quick pair of eyes on.

https://github.com/chef/chef-server/blob/6d76acee54d46f4c96ca151259364ef169cd2ab3/src/chef-server-ctl/plugins/ha.rb#L31
https://github.com/chef/chef-server/blob/6d76acee54d46f4c96ca151259364ef169cd2ab3/omnibus/files/private-chef-cookbooks/private-chef/libraries/private_chef.rb#L703

This can wait until after ChefConf, but I wanted to put it in the queue while I remembered.

**Comment by markan** [comment]
@chef/chef-server-maintainers This should be ready to merge; would love to have eyes on it.

File: omnibus/files/private-chef-cookbooks/private-chef/spec/libraries/private_chef_spec.rb [hunk 1]
```diff
@@ -165,7 +165,8 @@ def config_for(hostname)
     end
   end
 
-  context "in an HA topology" do
+# HA is deprecated
+  context "in an HA topology", focus: true do
```
Comment (lamont-granquist) [inline omnibus/files/private-chef-cookbooks/private-chef/spec/libraries/private_chef_spec.rb]: `focus: true` needs to be removed before merge

Comment (markan) [reply omnibus/files/private-chef-cookbooks/private-chef/spec/libraries/private_chef_spec.rb]: Crud. Thanks for spotting that.

**Comment by markan** [comment]
Passed CI as https://buildkite.com/chef/chef-chef-server-master-omnibus-adhoc/builds/188

Should be ready to merge.

File: omnibus/files/private-chef-cookbooks/private-chef/recipes/plugin_discovery.rb [hunk 2]
```diff
@@ -15,8 +15,8 @@
 # limitations under the License.
 #
 
-BACKUP_PLUGIN_LOCATIONS = %w(/opt/opscode/chef-server-plugin.rb
-                             /opt/chef-ha/chef-server-plugin.rb)
+# TODO Does this go away with HA?
```
Comment (stevendanna) [inline omnibus/files/private-chef-cookbooks/private-chef/recipes/plugin_discovery.rb]: I think it was also used by

- https://github.com/chef/omnibus-ha

which I think is also deprecated for a while now.  

The only other thing I can think of that *may* have been using it was the marketplace/opsworks stuff.  Maybe @ryancragun or @danielsdeleo can confirm that it isn't used there.

Comment (markan) [reply omnibus/files/private-chef-cookbooks/private-chef/recipes/plugin_discovery.rb]: I'm going to take this on in a separate PR.

Comment (markan) [reply omnibus/files/private-chef-cookbooks/private-chef/recipes/plugin_discovery.rb]: This is taken up in #1673

**Pull request merged**